pkg/updatechecker/updatechecker.go

```diff
@@ -752,25 +767,76 @@ func GetAvailableUpdates(kotsStore storepkg.Store, app *apptypes.App, license *k
 
 func isUpdateDeployable(updateCursor string, updates []upstreamtypes.Update) (bool, string) {
 	// iterate over updates in reverse since they are sorted in descending order
-	requiredUpdates := []upstreamtypes.Update{}
+	requiredUpdates := []string{}
 	for i := len(updates) - 1; i >= 0; i-- {
 		if updates[i].Cursor == updateCursor {
 			break
 		}
 		if updates[i].IsRequired {
-			requiredUpdates = append(requiredUpdates, updates[i])
+			requiredUpdates = append(requiredUpdates, updates[i].VersionLabel)
 		}
 	}
 	if len(requiredUpdates) > 0 {
-		versionLabels := []string{}
-		for _, v := range requiredUpdates {
-			versionLabels = append([]string{v.VersionLabel}, versionLabels...)
+		return false, getNonDeployableCause(requiredUpdates)
+	}
+	return true, ""
+}
+
+func getNonDeployableCause(requiredUpdates []string) string {
+	if len(requiredUpdates) == 0 {
+		return ""
+	}
+
+	versionLabels := []string{}
+	for _, versionLabel := range requiredUpdates {
+		versionLabels = append([]string{versionLabel}, versionLabels...)
+	}
+	versionLabelsStr := strings.Join(versionLabels, ", ")
+	if len(requiredUpdates) == 1 {
+		return fmt.Sprintf("This version cannot be deployed because version %s is required and must be deployed first.", versionLabelsStr)
+	}
+	return fmt.Sprintf("This version cannot be deployed because versions %s are required and must be deployed first.", versionLabelsStr)
+}
+
+func GetAvailableAirgapUpdates(app *apptypes.App, license *kotsv1beta1.License) ([]types.AvailableUpdate, error) {
+	updates := []types.AvailableUpdate{}
+	if err := filepath.Walk(AvailableUpdatesDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
 		}
-		versionLabelsStr := strings.Join(versionLabels, ", ")
-		if len(requiredUpdates) == 1 {
-			return false, fmt.Sprintf("This version cannot be deployed because version %s is required and must be deployed first.", versionLabelsStr)
+		if info.IsDir() {
+			return nil
+		}
+
+		airgapMetadata, err := kotsutil.FindAirgapMetaInBundle(path)
+		if err != nil {
+			return errors.Wrap(err, "failed to find airgap metadata")
+		}
+
+		if airgapMetadata.Spec.AppSlug != license.Spec.AppSlug && airgapMetadata.Spec.ChannelID != license.Spec.ChannelID {
```

Comment:
would probably be better to split them up
```suggestion
		if airgapMetadata.Spec.AppSlug != license.Spec.AppSlug || airgapMetadata.Spec.ChannelID != license.Spec.ChannelID {
```